src/Analyzers/CSharp/Analyzers/UseCollectionInitializer/CSharpObjectCreationHelpers.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseCollectionInitializer;
+
+internal static class CSharpObjectCreationHelpers
+{
+    public static bool IsInitializerOfLocalDeclarationStatement(
+        LocalDeclarationStatementSyntax localDeclarationStatement,
+        BaseObjectCreationExpressionSyntax rootExpression,
+        [NotNullWhen(true)] out VariableDeclaratorSyntax? variableDeclarator)
```

Comment:
crux of the change was that we had some suspect code working at the raw node level, trying to understand c# and vb nodes in a uniform way.  That code was simply broken.   We now defer to each language to handle this case as VB has some unique constructs we can't handle uniformly.

Reply:
this function is shared between UseCollectionInitializer and UseObjectInitializer, which is why it's a shared helper they can both use.